leadership.go

```diff
@@ -128,6 +132,7 @@ func (l *Lock) RefreshLock(ctx context.Context) error {
 
 	l.refreshFailures = 0
 	l.latestMetadataGeneration = attrs.Metageneration
+	l.latestGeneration = attrs.Generation
```

Comment:
Given we don't write anything to the object - we only write to the metadata - wouldn't the generation never change?